app/furniture/marketplace/product.rb

```diff
@@ -25,5 +27,15 @@ class Product < Record
     attribute :description, :string
 
     monetize :price_cents
+
+    before_commit :standardize_attachment_name
+
+    def standardize_attachment_name
+      return unless photo.attached? && photo.blob.persisted?
+      return if photo.blob.filename.to_s.start_with?(space.id.to_s)
+
+      new_name = "#{space.id}-#{photo.blob.filename}"
```

Comment:
It turns out that my naive approach to make this `space.id/...` didn't work, because Active Storage strips out the directories.

@zspencer is pre-pending the filenames with the space id (but having all files be together in the bucket) enough for our needs right now?

Reply:
I think so, I still think bucket-per-space is a better strategy for segmenting data but I can live with prefixing with space-id for now.

Reply:
PS. If we end up going with this approach, and liking it, I imagine that this callback method could be extracted into a module to be shared by any model that has attachments.